### PR TITLE
fix: ModelBuilder not passing HF_TOKEN to model.

### PR DESCRIPTION
**Pull request by makungaj1**
*Issue #, if available:*

*Description of changes:*
Fixes:
* ModelBuilder JS Builder not passing HF_TOKEN down to model.
* ModelBuilder build issue with model object.

*Testing done:*
Tested with this public [notebook](https://github.com/aws-samples/sagemaker-hosting/blob/main/SageMaker-Model-Builder/traditional-models/model-builder-xgboost.ipynb).

```
model_builder = ModelBuilder(
    model=target_model_id,
    env_vars={
        "HUGGING_FACE_HUB_TOKEN": hf_hub_token
    },
    schema_builder=SchemaBuilder(sample_input, sample_output),
    sagemaker_session=sagemaker_session,
    role_arn=my_role,
)
optimized_model = model_builder.optimize(
   instance_type="ml.inf2.8xlarge",
   compilation_config={
          "OverrideEnvironment": {
            "OPTION_TENSOR_PARALLEL_DEGREE": "2",
            "OPTION_N_POSITIONS": "8192",
            "OPTION_DTYPE": "fp16",
            "OPTION_ROLLING_BATCH": "auto",
            "OPTION_MAX_ROLLING_BATCH_SIZE": "4",
            "OPTION_NEURON_OPTIMIZE_LEVEL": "2",
         }
   },
   output_path=f"s3://{out_put_bucket}/{uuid.uuid4().hex}/"
)
predictor = optimized_model.deploy(initial_instance_count=1)
predictor.predict(sample_input)
predictor.delete_model()
predictor.delete_endpoint()
```

```
model_builder = ModelBuilder(
    model="meta-textgeneration-llama-3-70b",
    schema_builder=SchemaBuilder(sample_input, sample_output),
    sagemaker_session=sagemaker_session,
    role_arn=my_role,
)
model_builder.display_benchmark_metrics()
model_builder.set_deployment_config("lmi-optimized", "ml.p5.48xlarge")

optimized_model = model_builder.optimize(
    accept_eula=True,
    speculative_decoding_config={
        "ModelSource": draft_model_s3_uri
    },
    env_vars={
        "HF_TOKEN": hf_hub_token # verified that HF token was passed correctly
    },
)

predictor = optimized_model.deploy(initial_instance_count=1)
predictor.predict(sample_input)
predictor.delete_model()
predictor.delete_endpoint()
```

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/serve/builder/model_builder.py [hunk 1]
```diff
@@ -1256,8 +1258,10 @@ def _model_builder_optimize_wrapper(
             )
 
         if input_args:
+            print(input_args)
```
Comment (grenmester) [inline src/sagemaker/serve/builder/model_builder.py:1261]: is this supposed to be here? same for print below

File: src/sagemaker/serve/builder/jumpstart_builder.py [hunk 2]
```diff
@@ -757,8 +752,9 @@ def _optimize_for_jumpstart(
             if self.pysdk_model.deployment_config
             else None
         )
-        self.instance_type = instance_type or deployment_config_instance_type or _get_nb_instance()
-        self.role_arn = role_arn or self.role_arn
+        self.instance_type = (
```
Comment (grenmester) [inline src/sagemaker/serve/builder/jumpstart_builder.py:755]: should this be before `self.instance_type` is used in lines 714 and 719?

File: src/sagemaker/serve/builder/jumpstart_builder.py [hunk 3]
```diff
@@ -757,8 +752,9 @@ def _optimize_for_jumpstart(
...
+            self.instance_type or deployment_config_instance_type or _get_nb_instance()
```
Comment (grenmester) [inline src/sagemaker/serve/builder/jumpstart_builder.py:756]: can `self.instance_type` be not set before this function?

File: src/sagemaker/serve/builder/model_builder.py [hunk 4]
```diff
@@ -1256,8 +1258,10 @@ def _model_builder_optimize_wrapper(
...
             self.sagemaker_session.sagemaker_client.create_optimization_job(**input_args)
             job_status = self.sagemaker_session.wait_for_optimization_job(job_name)
+            print(job_status)
```
Comment (grenmester) [inline src/sagemaker/serve/builder/model_builder.py:1264]: is this `print` supposed to be here?

File: src/sagemaker/serve/builder/model_builder.py (hunk 1 above)
Comment (makungaj1) [reply src/sagemaker/serve/builder/model_builder.py:1261]: No, had it for debugging. I'll remove it.

**Review by grenmester — APPROVED** [review]

**Review by samruds — APPROVED** [review]

**Review by Aditi2424 — APPROVED** [review]

**Pull request merged**